### ISSUE TITLE
Accessors for extensions registered as decorated instances are typed Any

### ISSUE THREAD

**Issue by eskatos** [gradle/kotlin-dsl#932]
Accessors generated for extensions that were registered as decorated (synthetic type) instances are typed `Any`, preventing their easy configuration.

One example is the `eclipse` extension accessor.

### Expected Behavior

```kotlin
eclipse {
    classpath...
}
```

### Current Behavior

One has to either address the extension by type:

```kotlin
configure<EclipseModel> {
    classpath...
}
```

or cast:
```kotlin
eclipse {
    this as EclipseModel
    classpath...
}
```

**Comment by eskatos** [comment]
PR https://github.com/gradle/gradle/pull/5743

**Comment by eskatos** [comment]
PR merged, integrated in 0ded8fc092697798f634953dc6bd027c68e44efa
Closing as delivered
